servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/H2ClientParentChannelInitializer.java

```diff
@@ -84,4 +82,15 @@ public void channelRegistered(final ChannelHandlerContext ctx) {
             // Http2ConnectionHandler.processGoAwayWriteResult will close the connection after GO_AWAY is flushed
         }
     }
+
+    private static Http2Settings validateClientSettings(Http2Settings settings) {
+        final Boolean pushEnabled = settings.pushEnabled();
+        if (pushEnabled == null) {
+            // Notify server that this client does not support server push and request it to be disabled.
+            settings.pushEnabled(false);
+        } else if (pushEnabled) {
+            throw new IllegalArgumentException("push is enabled but not supported. settings=" + settings);
```

Comment:
We should also keep maxConcurrentStreams set to 0L and don't let users override it.